app/lib/features/events/sheets/edit_event_sheet.dart

```diff
@@ -250,116 +239,67 @@ class _EditEventSheetConsumerState extends ConsumerState<EditEventSheet> {
       confirmActionTitle: 'Save Changes',
       cancelActionTitle: 'Cancel',
       confirmActionOnPressed:
-          titleInput.isEmpty ? null : () => _handleUpdateEvent(context),
+          (titleInput.trim().isEmpty || startDate == null || endDate == null)
+              ? null
+              : _handleUpdateEvent,
       cancelActionOnPressed: () =>
           context.canPop() ? context.pop() : context.goNamed(Routes.main.name),
     );
   }
 
-  void _handleTitleChange(String? value) {
-    ref.read(_titleProvider.notifier).update((state) => value!);
-  }
-
-  void _handleUpdateEvent(BuildContext context) async {
-    showAdaptiveDialog(
-      barrierDismissible: false,
-      context: context,
-      builder: (context) => DefaultDialog(
-        title: Text(
-          'Updating Event',
-          style: Theme.of(context).textTheme.titleSmall,
-        ),
-        isLoader: true,
-      ),
-    );
-    final calendarEvent =
-        await ref.read(calendarEventProvider(widget.calendarId!).future);
+  void _handleUpdateEvent() async {
+    EasyLoading.show(status: 'Updating Event', dismissOnTap: false);
     try {
-      // initialize event update builder
-      final eventUpdateBuilder = calendarEvent.updateBuilder();
-
-      eventUpdateBuilder.title(ref.read(_titleProvider));
-
-      final date = ref.read(_dateProvider);
-      final startTime = ref.read(_startTimeProvider);
-      final utcStartDateTime = DateTime(
-        date.year,
-        date.month,
-        date.day,
-        startTime.hour,
-        startTime.minute,
-      ).toUtc();
-      eventUpdateBuilder
-          .utcStartFromRfc3339(utcStartDateTime.toIso8601String());
-
-      final endTime = ref.read(_endTimeProvider);
-      final utcEndDateTime = DateTime(
-        date.year,
-        date.month,
-        date.day,
-        endTime.hour,
-        endTime.minute,
-      ).toUtc();
-      eventUpdateBuilder.utcEndFromRfc3339(utcEndDateTime.toIso8601String());
-
-      eventUpdateBuilder.descriptionText(_descriptionController.text.trim());
-
-      final eventId = await eventUpdateBuilder.send();
-      debugPrint('Updated Calendar Event: ${eventId.toString()}');
-
-      // We are doing as expected, but the lints triggers.
-      // ignore: use_build_context_synchronously
-      if (!context.mounted) {
+      final asyncEvent = ref.read(calendarEventProvider(widget.calendarId!));
+      // We always have calendar object at this stage.
+      final startDate = ref.read(_startDateProvider);
+      final endDate = ref.read(_endDateProvider);
+      if (startDate == null && endDate == null) {
+        EasyLoading.showInfo(
+          'Please Select time',
+          duration: const Duration(seconds: 2),
```

Comment:
this is important. be noise and precise about it.
```suggestion
        EasyLoading.showError(
          'You must select a start and end date and time',
          duration: const Duration(seconds: 3),
```